### PR TITLE
[Deposit Summary] Show collapsed Deposit Summary until tapped

### LINKED ISSUE #11224: [Deposits Summary] Card expansion/contraction

### PR DESCRIPTION
**Pull request by joshheald**
Closes: #11224

## Description

The Deposits Summary is fairly large, and most of the time would require a user to scroll to do whatever they came to the Payments menu for.

The key information is the available and pending fund totals.

This PR collapses the view when the Payments Menu is loaded, and allows it to be expanded with a tap.

## Testing instructions

1. Launch the app
2. Switch to a WooPayments store with only one currency set up
3. Navigate to `Menu > Payments`
4. Observe that after a few seconds, the Deposit Summary is shown, collapsed.
5. Tap the expansion button
6. Observe that the Deposit Summary is shown in full

Repeat the above test with a multi-currency WooPayments account. (I can provide a login if needed)
Observe that the currency tabs are hidden when it's collapsed.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/2472348/c633581b-9705-4ca8-b116-30486eb7e2b5

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/Deposits Overview/WooPaymentsDepositsCurrencyOverviewView.swift [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                    Divider()
+
+                    HStack(alignment: .top) {
+                        Image(systemName: "calendar")
+                            .accessibilityHidden(true)
+                        Text(viewModel.balanceTypeHint)
+                            .font(.footnote)
+                    }
+                    .foregroundColor(.secondary)
+                    .padding(.vertical, 8)
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/Deposits Overview/WooPaymentsDepositsCurrencyOverviewView.swift:51]: Not necessarily now, but we can move the size-related numbers to a layout enum:
```
.padding(.vertical, 8)
[GridItem(.flexible(maximum: 70)
spacing: 16)
.padding(.vertical, 8)
```

**Review by iamgabrielma — COMMENTED** [review]
Looking great! Also works nicely in dark mode, landscape, and using larger accessibility fonts.

> Repeat the above test with a multi-currency WooPayments account. (I can provide a login if needed)

Would you mind to send me an invite to the multi-currency store? I couldn't test that bit. Approving it since it's feature flagged 🚢 

I found an oddity with the currency alignment, but I'm not sure where it comes from since my WooCommerce settings are set to `Currency Position: Left` and I cannot see anything in the View that could cause this. The currencies seem to show to the right only in the Deposits view, the rest of the app shows them correctly. It also happens when I log into the UK store in your recording, so seems to be on my end only 🤔 

| USD | Pound |
|--------|--------|
Simulator Screenshot - iPhone 15 - 2023-11-21 at 10 36 24 | Simulator Screenshot - iPhone 15 - 2023-11-21 at 10 37 45 |

Minor suggestion: When checking the view in landscape we may want to add additional leading margin: 
20231121_review_ui_landscape

**Comment by iamgabrielma** [comment]
I forgot, just curious: Since this would only be available for merchants that are in iOS 16+, and an empty view for earlier iOS, do we plan to add some disclaimer for these users? Or is just part of the current development but we'll tackle a view for these users well? eg: some sort of warning prompting to update. If we add nothing we should inform HEs so perhaps can be added to public documentation.

**Review by iamgabrielma — APPROVED** [review]

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/Deposits Overview/WooPaymentsDepositsCurrencyOverviewView.swift [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    init(viewModel: WooPaymentsDepositsCurrencyOverviewViewModel,
+         isExpanded: Binding<Bool>) {
+        self.viewModel = viewModel
+        self._isExpanded = isExpanded
+    }
+
+    var body: some View {
+        VStack {
+            Grid(alignment: .leading) {
+                GridRow {
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/Deposits Overview/WooPaymentsDepositsCurrencyOverviewView.swift:19]: Nice! 💯

**Comment by joshheald** [comment]
> Looking great! Also works nicely in dark mode, landscape, and using larger accessibility fonts.

Thanks for the review @iamgabrielma!
 
> Would you mind to send me an invite to the multi-currency store? I couldn't test that bit. 

I've sent you that now 😄 

> I found an oddity with the currency alignment, but I'm not sure where it comes from since my WooCommerce settings are set to `Currency Position: Left` and I cannot see anything in the View that could cause this. The currencies seem to show to the right only in the Deposits view, the rest of the app shows them correctly. It also happens when I log into the UK store in your recording, so seems to be on my end only 🤔

It's dependent on device locale settings, not Woo settings. Looking at it, I'm using a fairly vanilla currency formatter, so I'll revisit with a Woo specific one in a future PR. It may be a little interesting for the multi-currency case though, because the Woo formatter is likely only set up for the primary store currency. Things could get quite confusing for stores with two currencies that share a symbol, e.g. selling in both USD and CAD.

> Minor suggestion: When checking the view in landscape we may want to add additional leading margin: 

Thanks. I'll see whether I have time to fiddle with that later on in the week, but it seems like a very small deal ;)

**Comment by joshheald** [comment]
> I forgot, just curious: Since this would only be available for merchants that are in iOS 16+, and an empty view for earlier iOS, do we plan to add some disclaimer for these users? Or is just part of the current development but we'll tackle a view for these users well? eg: some sort of warning prompting to update. If we add nothing we should inform HEs so perhaps can be added to public documentation.

I don't think it's helpful to show a view saying "You don't get this" for older OSes, same as I don't think it makes sense to add that view for people using Stripe or other payment gateways. Since this isn't likely to be advertised anywhere, people just won't know what they don't have. 

I expect that we'll drop iOS 15 soon anyway, so it'll become a non-issue then. I'll happily inform HEs, but again, public documentation of this sort of minutae seems unneccesary. If/when we add Instant Deposits, that may be more relevant, but we could well have dropped iOS 15 by then anyway.

**Comment by iamgabrielma** [comment]
> It's dependent on device locale settings, not Woo settings. 

Oh that was it, it's locale settings indeed! 

> Looking at it, I'm using a fairly vanilla currency formatter, so I'll revisit with a Woo specific one in a future PR. It may be a little interesting for the multi-currency case though, because the Woo formatter is likely only set up for the primary store currency. Things could get quite confusing for stores with two currencies that share a symbol, e.g. selling in both USD and CAD.

Good catch.

> I expect that we'll drop iOS 15 soon anyway, so it'll become a non-issue then. I'll happily inform HEs, but again, public documentation of this sort of minutae seems unneccesary. If/when we add Instant Deposits, that may be more relevant, but we could well have dropped iOS 15 by then anyway.

That makes sense, thanks!

**Pull request merged**